descriptors/ModuleDescriptor-template.json

```diff
@@ -1,157 +1,167 @@
         {
-          "methods": [
-            "POST"
-          ],
-          "pathPattern": "/calendar/periods/{servicePointId}/period",
-          "permissionsRequired": [
-            "calendar.periods.item.post"
-          ]
+          "methods": ["GET"],
+          "pathPattern": "/calendar/calendars/{calendarId}",
+          "permissionsRequired": ["calendar.endpoint.calendars.calendarId.get"]
         },
         {
-          "methods": [
-            "GET"
-          ],
-          "pathPattern": "/calendar/periods/{servicePointId}/period/{periodId}",
-          "permissionsRequired": [
-            "calendar.periods.item.get"
-          ]
+          "methods": ["PUT"],
+          "pathPattern": "/calendar/calendars/{calendarId}",
+          "permissionsRequired": ["calendar.endpoint.calendars.calendarId.put"]
         },
         {
-          "methods": [
-            "PUT"
-          ],
-          "pathPattern": "/calendar/periods/{servicePointId}/period/{periodId}",
-          "permissionsRequired": [
-            "calendar.periods.item.put"
-          ]
+          "methods": ["DELETE"],
+          "pathPattern": "/calendar/calendars/{calendarId}",
+          "permissionsRequired": ["calendar.endpoint.calendars.calendarId.delete"]
         },
         {
-          "methods": [
-            "DELETE"
-          ],
-          "pathPattern": "/calendar/periods/{servicePointId}/period/{periodId}",
-          "permissionsRequired": [
-            "calendar.periods.item.delete"
-          ]
+          "methods": ["GET"],
+          "pathPattern": "/calendar/dates/{servicePointId}/surrounding-openings",
+          "permissionsRequired": ["calendar.endpoint.dates.get"]
         },
         {
-          "methods": [
-            "GET"
-          ],
-          "pathPattern": "/calendar/periods/{servicePointId}/calculateopening",
-          "permissionsRequired": [
-            "calendar.opening-hours.collection.get"
-          ]
+          "methods": ["GET"],
+          "pathPattern": "/calendar/dates/{servicePointId}/all-openings",
+          "permissionsRequired": ["calendar.endpoint.dates.get"]
         }
       ]
     },
     {
-      "id" : "_tenant",
-      "version" : "2.0",
-      "interfaceType" : "system",
-      "handlers" : [ {
-        "methods" : [ "POST" ],
-        "pathPattern" : "/_/tenant"
-      }, {
-        "methods" : [ "GET", "DELETE" ],
-        "pathPattern" : "/_/tenant/{id}"
-      } ]
+      "id": "_tenant",
+      "version": "2.0",
+      "interfaceType": "system",
+      "handlers": [
+        {
+          "methods": ["POST"],
+          "pathPattern": "/_/tenant"
+        },
+        {
+          "methods": ["GET", "DELETE"],
+          "pathPattern": "/_/tenant/{id}"
+        }
+      ]
     }
   ],
   "permissionSets": [
     {
-      "permissionName": "calendar.opening-hours.collection.get",
-      "displayName": "List calendar opening hours",
-      "description": ""
+      "permissionName": "calendar.view",
```

Comment:
Typically the back end modules do not define permission sets that are intended to be used by other modules.

Other back end modules are intended to only use the minimal permissions they can. And front end modules are intended to define the user assignable permission sets (view suggests it is a UI permission to me, at least based upon FOLIO's naming)

Reply:
I did not find too much guidance regarding permissions -- my intention was to provide granular permissions based off the endpoints (the ones with .endpoint, to address your other comment), then the set of CRUD-based ones to make it easier for other modules to consume/assign permissions (and be a little more future proof, if names need to change).

Then, consuming modules can use `calendar.view` and not worry about the nitty gritty of the actual endpoint names (and keeping those in sync with interface changes).  The UI module publishes the set of permissions meant to be assigned to the user, with similar CRUD names and simplified requiring permissions (e.g. `ui-calendar.view` can require `calendar.view`).

Reply:
> consuming modules can use calendar.view and not worry about the nitty gritty of the actual endpoint names (and keeping those in sync with interface changes).

In that situation, what is the value of the endpoint specific permissions?

I think this is a trade off between convenience for a consumer and the principle of least privilege.

Reply:
Least privilege is available, and in general each permission really only applies to one endpoint (just with nicer names), with the exception of view (which allows getting calendar info and calculating opening info for certain dates, but they're pretty similar and can be calculated/extrapolated from each other)